### PR TITLE
fix(recordings): fix flashing error state on first load

### LINKED ISSUE #12228: Recording playback sometimes flashes with unable to play
## Bug description

Seems like there is some incorrect logic that shows the "unable to play this recording" banner too early or incorrectly as very shortly after the recording plays. Thinking it might be some race condition or missed loading check.

Screenshot 2022-10-13 at 08 52 33

## How to reproduce

1. Tricky one to reproduce but clicking through a bunch of recordings it will eventually happen
2.
3.

## Environment

- [x] PostHog Cloud
- [ ] self-hosted PostHog, version/commit: please provide

### PR DESCRIPTION
**Pull request by alexkim205**
## Problem

Fixes https://github.com/PostHog/posthog/issues/12228

https://user-images.githubusercontent.com/13460330/195884235-55b6c326-ad56-4cef-8178-24aa622dc72d.mov

## Changes

https://user-images.githubusercontent.com/13460330/195884256-cceb8212-262f-4ffb-b809-4f16e4bf39da.mov

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

## How did you test this code?

Locally added a 2s await to `loadRecordingSnapshots` and saw that these changes fixed the bug.

### CODE REVIEW

File: frontend/src/scenes/session-recordings/player/sessionRecordingDataLogic.ts [hunk 1]
```diff
@@ -198,6 +198,15 @@ export const sessionRecordingDataLogic = kea<sessionRecordingDataLogicType>([
...
         ],
+        sessionPlayerDataLoading: [
+            false,
+            {
+                loadRecordingMetaSuccess: (_, { sessionPlayerData }) => {
+                    // Keep `sessionPlayerDataLoading` for when meta is loaded before snapshots finish loading
+                    return Object.keys(sessionPlayerData?.snapshotsByWindowId ?? {}).length === 0
+                },
+            },
+        ],
```
Comment (benjackwhite) [inline frontend/src/scenes/session-recordings/player/sessionRecordingDataLogic.ts:209]: Not sure how keen on this I am. We're overriding the built in loader reducer which feels error prone. Plus we could have a reverse race condition where it happens the other way.

I think we either need to break out the two loaders (one for metadata, one for snapshots) or have a unique reducer that records both loading states together so we can tell if we are "loadingMetadataOrSnapshots"

**Review by benjackwhite — DISMISSED** [review]
I think we should take the extra step and do a bit more work for a proper fix as this feels a little error prone

File: frontend/src/scenes/session-recordings/player/sessionRecordingDataLogic.ts (hunk 1 above)
Comment (alexkim205) [reply frontend/src/scenes/session-recordings/player/sessionRecordingDataLogic.ts:209]: True, this solution does rely on snapshots always taking a longer time. I'll go with the breaking up into two loaders method

**Review by benjackwhite — APPROVED** [review]
Much better! I did a couple of small fixes to some typings and also moved the bufferedTo into the selector code as it is purely a derived value.

**Pull request merged**